testfuzz/src/main/java/org/hyperledger/besu/testfuzz/P256VerifyFuzzTarget.java

```diff
@@ -0,0 +1,911 @@
+  "NarrowingCompoundAssignment"
+})
+public class P256VerifyFuzzTarget implements FuzzTarget {
+
+  private static final Logger LOG = LoggerFactory.getLogger(P256VerifyFuzzTarget.class);
+
+  // SECP256R1 curve parameters
+  private static final X9ECParameters R1_PARAMS = SECNamedCurves.getByName("secp256r1");
+  private static final BigInteger N = R1_PARAMS.getN(); // Order of the curve
+  private static final BigInteger P =
+      R1_PARAMS.getCurve().getField().getCharacteristic(); // Field prime
+  private static final ECPoint G = R1_PARAMS.getG(); // Generator point
+
+  // Expected verification results
+  private static final Bytes32 VALID_SIGNATURE_RESULT = Bytes32.leftPad(Bytes.of(1), (byte) 0);
+
+  // Core components
+  private final P256VerifyPrecompiledContract verificationContractNative;
+  private final P256VerifyPrecompiledContract verificationContractJava;
+  private final MessageFrame messageFrame;
+  private final SignatureAlgorithm javaSignatureAlgorithm;
+  private final SignatureAlgorithm nativeSignatureAlgorithm;
+  private final SecureRandom random;
+
+  // Mutation strategies for comprehensive testing
+  private enum MutationStrategy {
+    NO_MUTATION, // Use valid signature as-is
+    BIT_FLIP, // Flip random bits in the signature
+    BOUNDARY_VALUES, // Test boundary conditions (0, n-1, n, p-1, p)
+    ARITHMETIC_MUTATION, // Apply arithmetic operations to signature components
+    CURVE_ATTACKS, // Test curve-specific edge cases
+    SIGNATURE_MALLEABILITY, // Test signature malleability (s vs n-s)
+    POINT_MANIPULATION, // Manipulate public key coordinates
+    BOUNCYCASTLE_BYPASS // Test BouncyCastle validation bypass scenarios
+  }
+
+  /** Initializes the fuzz target with both Java and Native signature implementations. */
+  public P256VerifyFuzzTarget() {
+    this.messageFrame = mock(MessageFrame.class);
+
+    // Create separate instances for Java and Native signing
+    this.javaSignatureAlgorithm = createJavaSignatureAlgorithm();
+    this.nativeSignatureAlgorithm = createNativeSignatureAlgorithm();
+
+    // Create separate precompile instances for Java and Native verification
+    this.verificationContractNative =
+        new P256VerifyPrecompiledContract(new OsakaGasCalculator(), this.nativeSignatureAlgorithm);
+    this.verificationContractJava =
+        new P256VerifyPrecompiledContract(new OsakaGasCalculator(), this.javaSignatureAlgorithm);
+
+    this.random = new SecureRandom();
+  }
+
+  /** Creates a SECP256R1 instance configured to use Java implementation. */
+  private SignatureAlgorithm createJavaSignatureAlgorithm() {
+    SECP256R1 algo = new SECP256R1();
+    algo.disableNative(); // Force Java implementation
+    return algo;
+  }
+
+  /** Creates a SECP256R1 instance configured to use Native implementation. */
+  private SignatureAlgorithm createNativeSignatureAlgorithm() {
+    SECP256R1 algo = new SECP256R1();
+    algo.maybeEnableNative(); // Try to enable native implementation
```

Comment:
There's a bit of a quirk here actually.

There's actually three implementations:
1. Java (via SignatureAlgorithm)
2. Native (via SignatureAlgorithm) which this line is enabling
3. Native (via BoringSSL) which is the fastest, desirable implementation.

The unit tests test all three.

P256VerifyBenchmark can switch between "--native" (BoringSSL) and "--nonative" (Java), so (2) is ignored.

I think it is quite unlikely a user will run with (2) in reality. I think typically they have all the native libs or they have none.

Your tests here do seem to only care about BoringSSL versus Java which I think is good. However, that makes this line about (2) a bit redundant/confusing.